### PR TITLE
Refactor address book to nursey book

### LINKED ISSUE #200: Refactor `addressBook` to `nurseyBook`

### PR DESCRIPTION
**Pull request by tsy24**
Resolves #200

### CODE REVIEW

File: src/main/java/nurseybook/commons/util/StringUtil.java [hunk 1]
```diff
@@ -28,8 +27,8 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         requireNonNull(word);
 
         String preppedWord = word.trim();
-        checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
-        checkArgument(preppedWord.split("\\s+").length == 1, "Word parameter should be a single word");
+        AppUtil.checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
```
Comment (xiongjya) [inline src/main/java/nurseybook/commons/util/StringUtil.java:30]: Could you import this at the start of the file and simply use `checkArgument(...)` instead? Similarly for line 31 below.

File: src/main/java/nurseybook/logic/parser/AddTagCommandParser.java [hunk 2]
```diff
@@ -23,14 +22,14 @@
      */
     public AddTagCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_TAG);
```
Comment (xiongjya) [inline src/main/java/nurseybook/logic/parser/AddTagCommandParser.java:25]: Similarly in this file, could you import the `CliSyntax` before with `import static nurseybook.logic.parser.CliSyntax.PREFIX_TAG;` instead?

File: src/main/java/nurseybook/logic/parser/EditTaskCommandParser.java [hunk 3]
```diff
@@ -1,23 +1,18 @@
-package seedu.address.logic.parser;
+package nurseybook.logic.parser;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
```
Comment (xiongjya) [inline src/main/java/nurseybook/logic/parser/EditTaskCommandParser.java:5]: Similarly, here with `import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;` - would it be better?

File: src/main/java/nurseybook/logic/parser/FilterCommandParser.java [hunk 4]
```diff
@@ -1,15 +1,14 @@
-package seedu.address.logic.parser;
+package nurseybook.logic.parser;
 
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
```
Comment (xiongjya) [inline src/main/java/nurseybook/logic/parser/FilterCommandParser.java:4]: `CliSyntax` for this file too!

File: src/main/java/nurseybook/model/ModelManager.java [hunk 5]
```diff
@@ -34,11 +34,11 @@
     /**
      * Initializes a ModelManager with the given versionedNurseyBook and userPrefs.
      */
-    public ModelManager(ReadOnlyAddressBook versionedNurseyBook, ReadOnlyUserPrefs userPrefs) {
+    public ModelManager(ReadOnlyNurseyBook versionedNurseyBook, ReadOnlyUserPrefs userPrefs) {
         super();
-        requireAllNonNull(versionedNurseyBook, userPrefs);
+        CollectionUtil.requireAllNonNull(versionedNurseyBook, userPrefs);
```
Comment (xiongjya) [inline src/main/java/nurseybook/model/ModelManager.java:39]: If you have imported `CollectionUtil` at the start, will using it here in the prefix be necessary?

Recurring in lines 157, 172

File: src/main/java/nurseybook/model/task/Task.java [hunk 6]
```diff
@@ -171,7 +170,7 @@ public Task markAsNotOverdue() {
     public Task updateDateRecurringTask() {
         LocalDateTime currentDateTime = LocalDateTime.now();
         if (recurrence.isRecurring()) {
-            RecurrenceType recurrenceType = recurrence.getRecurrenceType();
+            Recurrence.RecurrenceType recurrenceType = recurrence.getRecurrenceType();
```
Comment (xiongjya) [inline src/main/java/nurseybook/model/task/Task.java:173]: Would it be better to import `Recurrence.RecurrenceType` at the start?

File: src/test/java/nurseybook/logic/LogicManagerTest.java [hunk 7]
```diff
@@ -48,23 +47,23 @@
...
-        StorageManager storage = new StorageManager(addressBookStorage, userPrefsStorage);
+        StorageManager storage = new StorageManager(nurseyBookStorage, userPrefsStorage);
         logic = new LogicManager(model, storage);
     }
 
     @Test
     public void execute_invalidCommandFormat_throwsParseException() {
         String invalidCommand = "uicfhmowqewca";
-        assertParseException(invalidCommand, MESSAGE_UNKNOWN_COMMAND);
+        assertParseException(invalidCommand, Messages.MESSAGE_UNKNOWN_COMMAND);
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/LogicManagerTest.java:60]: Would it be better to import the `Messages` file at the top of this file instead?

Recurring in ine 66.

File: src/test/java/nurseybook/logic/commands/AddCommandIntegrationTest.java [hunk 8]
```diff
@@ -23,7 +23,7 @@
 
     @BeforeEach
     public void setUp() {
-        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        model = new ModelManager(TypicalElderlies.getTypicalNurseyBook(), new UserPrefs());
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/AddCommandIntegrationTest.java:26]: Would it be better to import `TypicalElderlies` at the start of the file?

File: src/test/java/nurseybook/logic/commands/AddCommandTest.java [hunk 9]
```diff
@@ -1,31 +1,32 @@
...
+import nurseybook.testutil.ElderlyBuilder;
+import nurseybook.testutil.ModelStub;
+import nurseybook.testutil.NurseyBookBuilder;
 
 public class AddCommandTest {
 
     @Test
     public void constructor_nullElderly_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new AddCommand(null));
+        Assert.assertThrows(NullPointerException.class, () -> new AddCommand(null));
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/AddCommandTest.java:29]: Would it be better to import `Assert` at the start of the file?

File: src/test/java/nurseybook/logic/commands/AddTagCommandTest.java [hunk 10]
```diff
@@ -1,33 +1,30 @@
...
+import nurseybook.model.NurseyBook;
+import nurseybook.model.UserPrefs;
+import nurseybook.model.person.Elderly;
+import nurseybook.testutil.ElderlyBuilder;
+import nurseybook.testutil.TypicalElderlies;
+import nurseybook.testutil.TypicalIndexes;
 public class AddTagCommandTest {
 
-    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model model = new ModelManager(TypicalElderlies.getTypicalNurseyBook(), new UserPrefs());
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/AddTagCommandTest.java:27]: Similarly, importing `TypicalElderlies` at the start of the file would be better?

Recurring in lines 48, 52.

File: src/test/java/nurseybook/logic/commands/AddTagCommandTest.java [hunk 11]
```diff
@@ -82,7 +79,7 @@ public void execute_addExistingTagUnfilteredList_success() {
 
     @Test
     public void execute_addTagFilteredList_success() {
-        showElderlyAtIndex(model, INDEX_FIRST);
+        CommandTestUtil.showElderlyAtIndex(model, INDEX_FIRST);
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/AddTagCommandTest.java:82]: Similarly, import `CommandTestUtil` at the top of the file would be better?

Recurring in lines 104, 113, 120.

File: src/test/java/nurseybook/logic/commands/ClearCommandTest.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+        Model expectedModel = new ModelManager();
+        expectedModel.commitNurseyBook(new CommandResult(ClearCommand.MESSAGE_SUCCESS));
+
+        assertCommandSuccess(new ClearCommand(), model, ClearCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_nonEmptyNurseyBook_success() {
+        Model model = new ModelManager(TypicalElderlies.getTypicalNurseyBook(), new UserPrefs());
+        Model expectedModel = new ModelManager(TypicalElderlies.getTypicalNurseyBook(), new UserPrefs());
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/ClearCommandTest.java:27]: Import `TypicalElderlies` at the start of the file would be better?

File: src/test/java/nurseybook/logic/commands/CommandTestUtil.java [hunk 13]
```diff
@@ -184,12 +183,13 @@ public static void showElderlyAtIndex(Model model, Index targetIndex) {
...
 
     /**
-     * Deletes the first person in {@code model}'s filtered list from {@code model}'s address book.
+     * Deletes the first person in {@code model}'s filtered list from {@code model}'s nursey book.
      */
     public static CommandResult deleteFirstElderly(Model model) {
         Elderly firstElderly = model.getFilteredElderlyList().get(0);
         model.deleteElderly(firstElderly);
-        CommandResult result = new CommandResult(String.format(MESSAGE_DELETE_ELDERLY_SUCCESS, firstElderly));
+        CommandResult result = new CommandResult(String.format(DeleteCommand.MESSAGE_DELETE_ELDERLY_SUCCESS,
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/CommandTestUtil.java:191]: Would importing `DeleteCommand` at the start of the file be better?

File: src/test/java/nurseybook/logic/commands/DeleteCommandTest.java [hunk 14]
```diff
@@ -1,35 +1,34 @@
...
 
-    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model model = new ModelManager(TypicalElderlies.getTypicalNurseyBook(), new UserPrefs());
 
     @Test
     public void execute_validIndexUnfilteredList_success() {
-        Elderly elderlyToDelete = model.getFilteredElderlyList().get(INDEX_FIRST.getZeroBased());
-        DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST);
+        Elderly elderlyToDelete = model.getFilteredElderlyList().get(TypicalIndexes.INDEX_FIRST.getZeroBased());
+        DeleteCommand deleteCommand = new DeleteCommand(TypicalIndexes.INDEX_FIRST);
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/DeleteCommandTest.java:31]: Recurring in lines 26, 30, 31, 52, and so on - would it be better to import such needed files at the top of the file?

File: src/test/java/nurseybook/logic/commands/DeleteNokCommandTest.java [hunk 15]
```diff
@@ -1,38 +1,37 @@
...
 
+import static nurseybook.logic.commands.CommandTestUtil.assertCommandFailure;
+import static nurseybook.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static nurseybook.logic.commands.CommandTestUtil.showElderlyAtIndex;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
-import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
-import static seedu.address.logic.commands.CommandTestUtil.showElderlyAtIndex;
-import static seedu.address.testutil.TypicalElderlies.getTypicalAddressBook;
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/DeleteNokCommandTest.java:8]: Missing imports in this file as well that would be better to include at the start!

File: src/test/java/nurseybook/logic/commands/DeleteTagCommandTest.java [hunk 16]
```diff
@@ -1,32 +1,31 @@
...
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static seedu.address.logic.commands.CommandTestUtil.SET_ONE_TAG;
-import static seedu.address.logic.commands.CommandTestUtil.SET_TWO_TAGS;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_DIABETES;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_FRIEND;
-import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
-import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
-import static seedu.address.logic.commands.CommandTestUtil.showElderlyAtIndex;
-import static seedu.address.testutil.TypicalElderlies.getTypicalAddressBook;
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/DeleteTagCommandTest.java:12]: Missing imports in this file as well that would be better to include at the start!

File: src/test/java/nurseybook/logic/commands/DeleteTaskCommandTest.java [hunk 17]
```diff
@@ -1,29 +1,27 @@
-package seedu.address.logic.commands;
+package nurseybook.logic.commands;
```
Comment (xiongjya) [inline src/test/java/nurseybook/logic/commands/DeleteTaskCommandTest.java:1]: Missing imports in this file as well that would be better to include at the start!

File: src/test/java/nurseybook/testutil/TypicalTasks.java [hunk 18]
```diff
@@ -1,89 +1,90 @@
...
+import nurseybook.model.task.Recurrence;
+import nurseybook.model.task.Task;
 
 public class TypicalTasks {
 
     public static final Task KEITH_INSULIN = new TaskBuilder().withDesc(VALID_DESC_MEDICINE)
             .withDateTime(VALID_DATE_NOV, VALID_TIME_SEVENPM).withNames(VALID_NAME_KEITH)
             .withStatus("false", "true")
-            .withRecurrence(RecurrenceType.NONE.name())
+            .withRecurrence(Recurrence.RecurrenceType.NONE.name())
```
Comment (xiongjya) [inline src/test/java/nurseybook/testutil/TypicalTasks.java:27]: Missing imports in this file as well that would be better to include at the start!

**Review by xiongjya — COMMENTED** [review]
Did not manage to review all the files - but the main issue will be missing imports. But it is just for standardisation sake - and the rest looks good!

**Review by nicole-luo-exe — APPROVED** [review]
LGTM

**Pull request merged**